spec/controllers/dashboard_controller_spec.rb

```diff
@@ -340,4 +341,8 @@ def expect_failed_login(flash = nil)
     expect(response.body).to match(/flash_msg_div/) if flash
     # expect(result.body.to match(/flash_msg_div.*replaceWith.*#{msg}/) if flash
   end
+
+  def browser_info(typ)
+    session.fetch_path(:browser, typ).to_s
```

Comment:
The real session isn't actually a hash, so I don't think it'll have `fetch_path`.. this is probably inadvertently relying on the fact that the session in the spec *is* a hash. Given that context, perhaps we'd be better/safer with a straight `session[:browser][:version]`?

(Disclaimer: I haven't checked either of the above assertions about what is / is not a hash.. they're just from memory, and may be blatant lies.)